### PR TITLE
Refactor SMS

### PR DESCRIPTION
**Pull request by natebrennand**
Figured out how to simplify the Account logic! All packages will import the `act` package then anonymously embed the `Account` struct. The types that are embedding it inherit it's methods automagically.
Worked fine for sms & shortcodes so it should for everything else.

Also added `golint` and `go vet` to the travis build. Currently they're limited to sms & shortcodes because the linter will fail on every other package.

Changes to sms:
- got rid of unnecessary private methods. (involved ripping out some tests)
- tried a different style of keeping track of URL's, I think it keeps the namespace a bit cleaner
- standardized doing project imports prior to stdlib imports

### CODE REVIEW

File: account.go [hunk 1]
```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/natebrennand/twiliogo/sms/shortcodes"
 	"github.com/natebrennand/twiliogo/transcription"
 	"github.com/natebrennand/twiliogo/voice"
+
```
Comment (chelpu) [inline account.go]: ew

File: sms/callback.go [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package sms
 
 import (
-	"fmt"
 	"github.com/natebrennand/twiliogo/common"
+
```
Comment (chelpu) [inline sms/callback.go]: WHITESPACE

Comment (natebrennand) [inline sms/callback.go]: IT'S GONE

File: account.go (hunk 1 above)
Comment (natebrennand) [inline account.go]: gross

File: sms/sms.go [hunk 3]
```diff
@@ -60,7 +90,7 @@ func (p Post) GetReader() io.Reader {
 	return strings.NewReader(v.Encode())
 }
 
-// Validates the Voice Post to ensure validity.
+// Validate the Voice Post to ensure validity.
```
Comment (chelpu) [inline sms/sms.go:92]: dem changes

**Comment by chelpu** [comment]
+1 :thumbsup:

**Pull request merged**